speakcivi.php

```diff
@@ -131,6 +130,85 @@ function speakcivi_civicrm_tokenValues(&$values, $cids, $job = null, $tokens = a
   foreach ($cids as $cid) {
     $values[$cid]['speakcivi.confirmation_hash'] = sha1(CIVICRM_SITE_KEY . $cid);
   }
+
+  //  cargo cult wants me to use this too: || in_array('member_city', $tokens['speakcivi']
+  if (array_key_exists('member_city', $tokens['speakcivi'])) {
+    speakcivi_member_city($values, $cids, $context);
+  }
+}
+
+
+function speakcivi_member_city(&$values, $cids, $context) {
+    /* 
+    if ($context ) {
+        foreach($context as $k => $v) {
+            CRM_Core_Error::debug_log_message("Context has $k $v");
+        }
+    }
+    else {
+            CRM_Core_Error::debug_log_message("Context is NULL");
+    }
+    */
+	$dao = CRM_Utils_SQL_Select::from('civicrm_contact cc')
+		->join('ca', 'LEFT JOIN civicrm_address ca on cc.id=ca.contact_id and ca.is_primary=1')
+		->join('country', 'LEFT JOIN civicrm_country country on country.id=country_id')
+		->select('contact_id, preferred_language, city, country.iso_code')
+        ->where('cc.id IN (#cids)', array('cids' => $cids))
+        ->execute();
+
+    $FALLBACK_CITY_VALUES = array(
```

Comment:
Can you move that to CRM/Speakcivi/Tools/Dictionary.php ?
I don't like adding more stuff there, but I guess that will do until we create a proper locale file for Speakcivi.